### ISSUE TITLE
allow to configure s3 subpath (bucket prefix configuration)

### ISSUE THREAD

**Issue by oneacl** [grafana/loki#5889]
Is your feature request related to a problem? Please describe.

Currently a full S3 bucket is required for Loki - however, in some situations a bucket may need to be shared with other apps (one example would be aws lightsail where there's a fixed cost per bucket).

Describe the solution you'd like

Ability to specify a folder where all Loki data will be stored in a bucket

I've raised a similar issue with Mimir: https://github.com/grafana/mimir/issues/1682

Thank you.

**Comment by arcosx** [comment]
I think this issue is very important and I am facing the same problem.

**Comment by arcosx** [comment]
Can this issue be assigned to me? I plan to implement a subpath function for object storage (s3,gcs,azure storage....). 
I can give a proposal. @kavirajk @owen-d

**Comment by fzyzcjy** [comment]
Facing the same problem. This is very much needed since we are using that bucket for other purposes as well.

**Comment by Sylphe88** [comment]
Need the same feature here, so that I can fit all the observability backends in the same bucket (thanos, loki...)!

**Comment by theprometeus** [comment]
Bumping it cause we are having the same problem.

**Comment by zerda** [comment]
Loki communicates with the S3 bucket using the [`BucketClient`](https://github.com/grafana/loki/blob/main/pkg/storage/bucket/s3/bucket_client.go) from `github.com/thanos-io/thanos/pkg/objstore/s3`.

Since https://github.com/thanos-io/thanos/pull/5337 has been merged, implementing bucket prefix in loki should be much simpler.

**Comment by oneacl** [comment]
@dimitarvdimitrov - thank you for https://github.com/grafana/mimir/pull/1686

was wondering if there are any plans to retrofit this into loki as well?

Thank you.

**Comment by nicon89** [comment]
Is there any update on this?

**Comment by davinkevin** [comment]
+1000 on this one! Really important when you want to backup a complete `monitoring` bucket with all [metrics|traces|logs] in the same state.

**Comment by ningyougang** [comment]
@matschaffer-roblox did you built the loki image by yourself? which loki image did you use?

**Comment by dragonball-wooboo** [comment]
Is there any update on this one?

**Comment by rubenvw-ngdata** [comment]
Also interested in how this is done for the helm chart. Is it documented somewhere?

**Comment by adil-jeel** [comment]
Is there any update on this one?

**Comment by awoimbee** [comment]
Hi, I'm a bit lost, I want to use this feature that should be available since #10096 got merged ([it's in the code](https://github.com/grafana/loki/blob/a6d1e7bdbbf6b4ab34ca46e70267d6ea450aa3cc/pkg/storage/factory.go#L330)) but with loki `2.9.4` I get:
```
failed parsing config: /etc/loki/config/config.yaml: yaml: unmarshal errors:
  line 71: field object_prefix not found in type storage.Config
```
I set `loki.storage_config.object_prefix` to `loki/` in the helm chart, it translates in the `loki` configmap to `storage_config.object_prefix`.

**Comment by tculp** [comment]
On that note, this feature has now been merged to main for nearly 5 whole months but not included in 2.9.3, 2.9.4, or 2.9.5. Could this please get included in a release, and also updated in the helm chart, added to documentation, etc?

@slim-bean I see you're the one who merged the original feature, so I'm tagging you for visibility
